R/bundlePackage.R

```diff
@@ -25,135 +25,59 @@ bundlePackages <- function(appDir,
   )
   deps <- snapshotRDependencies(appDir, inferredRDependencies, verbose = verbose)
 
-  packageMessages <- c()
-  errorMessages <- c()
-  packages <- list()
-
-  # construct package list from dependencies
-  for (i in seq_len(nrow(deps))) {
-    name <- deps[i, "Package"]
-
-    # get package info
-    info <- as.list(deps[i, c("Source", "Repository")])
-
-    # include github package info
-    info <- c(info, as.list(deps[i, grep("Github", colnames(deps), perl = TRUE, value = TRUE)]))
```

Comment:
Does this imply that rsconnect doesn't actually support packages installed from GitLab and Bitbucket? Or does connect read the needed metadata from the serialised DESCRIPTION and these extra columns can be dropped?

Reply:
Can you point me to the code in connect that parses the packages info from the manifest?

Reply:
I believe that this is consumed by shinyapps.io, not Connect. I'll try to uncover the reference and send it separately.

Reply:
Oh, what does connect use? Just the lockfile?

Reply:
Connect prefers the packrat/packrat.lock. When only a manifest is provided, a lockfile is constructed from the manifest before handing off to packrat for package installation.

Reply:
Ok, good to know that this is all for shinyapps.io. I'll definitely add some comments.

Reply:
At one point in time, there were two packages. https://github.com/rstudio/shinyapps

Reply:
And ShinyApps.io only supports GitHub packages, hence the filtering.

Reply:
Discussion now captured in comments.